examples/light-switch-app/silabs/src/AppTask.cpp

```diff
@@ -143,6 +143,8 @@ void AppTask::SwitchActionEventHandler(AppEvent * aEvent)
             mCurrentButtonState ? LightSwitchMgr::LightSwitchAction::On : LightSwitchMgr::LightSwitchAction::Off;
 
         LightSwitchMgr::GetInstance().TriggerLightSwitchAction(action);
+        LightSwitchMgr::GetInstance().currentLevel = mCurrentButtonState ? MAX_LEVEL : MIN_LEVEL;
+        LightSwitchMgr::GetInstance().TriggerLevelControlAction(LightSwitchMgr::GetInstance().currentLevel);
```

Comment:
Why are we changing the level of the light when turning it on or off? This doesn't seem right...

Reply:
When the light is Off, the level of the light is minimum. 
When the light is On, the level of the light is maximum. 
I'm changing the level here as it is mentioned in design doc that the level and on-off attributes will be adjusted.

Reply:
This doesn't integrate correctly with the level control - please check the level control implementation.